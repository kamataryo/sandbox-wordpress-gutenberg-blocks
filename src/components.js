export const edit = ({ attributes, setAttributes }) => {
  const onChange = event => setAttributes({ text: event.target.value })
  return (
    <div>
      <input value={attributes.text} onChange={onChange} type={'text'} />
    </div>
  )
}

export const save = ({ attributes }) => {
  return (
    <div>
      <span className={'my-text'}>{`saved: ${attributes.text}`}</span>
    </div>
  )
}
