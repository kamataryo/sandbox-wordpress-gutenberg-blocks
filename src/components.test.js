import { shallow, mount } from 'enzyme'
import { edit as Edit, save as Save } from './components'

describe('test for Edit component', () => {
  const attributes = { text: '' }

  test('render input', () => {
    const wrapper = shallow(<Edit attributes={attributes} />)
    expect(wrapper.find('input').length).toBe(1)
  })

  test('match snapshot', () => {
    const dom = mount(<Edit attributes={attributes} />).html()
    expect(dom).toMatchSnapshot()
  })
})

describe('test for Save component', () => {
  const attributes = { text: 'abc' }

  test('render text', () => {
    const wrapper = shallow(<Save attributes={attributes} />)
    expect(wrapper.find('.my-text').text()).toEqual('saved: abc')
  })

  test('match snapshot', () => {
    const dom = mount(<Save attributes={attributes} />).html()
    expect(dom).toMatchSnapshot()
  })
})
