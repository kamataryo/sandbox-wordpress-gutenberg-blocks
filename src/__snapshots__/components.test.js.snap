// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Edit component match snapshot 1`] = `"<div><input type=\\"text\\" value=\\"\\"></div>"`;

exports[`test for Save component match snapshot 1`] = `"<div><span class=\\"my-text\\">saved: abc</span></div>"`;
